rustls/src/server/hs.rs

```diff
@@ -348,6 +348,16 @@ impl ExpectClientHello {
             cx.common.protocol,
         );
 
+        // And supported kx groups
+        let suitable_suites = suites::reduce_given_kx_groups(
+            &suitable_suites,
+            client_hello.get_namedgroups_extension(),
+            &self
+                .config
+                .provider
+                .supported_kx_group_names(),
+        );
+
```

Comment:
I think you should add something like this here if we meet the case mentioned in the RFC at the end of the paragraph 1 : https://datatracker.ietf.org/doc/html/rfc7919#section-4
```rust
cx.common.send_fatal_alert(
  AlertDescription::InsufficientSecurity,
  PeerIncompatible::NoKxGroupsInCommon,  // or another newly added error varaint under `PeerIncompatible`
)
```